packages/block-library/src/block/edit.js

```diff
@@ -256,27 +292,18 @@ function ReusableBlockEdit( {
 
 	return (
 		<>
-			{ userCanEdit && onNavigateToEntityRecord && (
-				<BlockControls>
-					<ToolbarGroup>
-						<ToolbarButton onClick={ handleEditOriginal }>
-							{ __( 'Edit original' ) }
-						</ToolbarButton>
-					</ToolbarGroup>
-				</BlockControls>
-			) }
-
-			{ canOverrideBlocks && (
-				<BlockControls>
-					<ToolbarGroup>
-						<ToolbarButton
-							onClick={ resetContent }
-							disabled={ ! content }
-						>
-							{ __( 'Reset' ) }
-						</ToolbarButton>
-					</ToolbarGroup>
-				</BlockControls>
+			{ hasResolved && (
+				<ReusableBlockControl
+					recordId={ ref }
+					canOverrideBlocks={ canOverrideBlocks }
+					hasContent={ !! content }
+					handleEditOriginal={
+						onNavigateToEntityRecord
+							? handleEditOriginal
+							: undefined
+					}
+					resetContent={ resetContent }
+				/>
```

Comment:
Unrelated change showcases the benefits of this proposal. I am happy to extract it into a separate PR.

**Requests before with four pattern blocks on canvas:**
![CleanShot 2024-07-15 at 16 52 13](https://github.com/user-attachments/assets/884e8fb7-6bff-4c5d-bfc6-ed5e46886433)


**Requests after:**
![CleanShot 2024-07-15 at 16 52 36](https://github.com/user-attachments/assets/29505301-9572-4dd0-bbdd-d405d3f80528)